L1Trigger/L1TTrackMatch/plugins/L1TkElectronTrackProducer.cc

```diff
@@ -0,0 +1,340 @@
+// -*- C++ -*-
+//
+/**\class L1TkElectronTrackMatchAlgo
+
+ Description: Producer of a TkElectron, for the algorithm matching a L1Track to the L1EG object
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/L1TCorrelator/interface/TkElectron.h"
+#include "DataFormats/L1TCorrelator/interface/TkElectronFwd.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+// Matching Algorithm
+#include "L1Trigger/L1TTrackMatch/interface/L1TkElectronTrackMatchAlgo.h"
+#include "L1Trigger/L1TTrackMatch/interface/L1TkElectronEtComparator.h"
+#include "L1Trigger/L1TTrackMatch/interface/pTFrom2Stubs.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+
+#include <string>
+#include "TMath.h"
+
+using namespace l1t;
+
+//
+// class declaration
+//
+
+class L1TkElectronTrackProducer : public edm::EDProducer {
```

Comment:
legacy producers are not allowed, use modern producers (everywhere)

Reply:
done